### ISSUE TITLE
Namespace and type name are unknown if test case name contains backslash

### ISSUE THREAD

**Issue by tkohlmeier** [spekt/nunit.testlogger#66]
If a test case name contains a backslash (this can be achived with TestCaseSource tests), the NUnit report lists theses tests under the namespace and type `Unknown`.

Maybe the backslash must be escaped?

**Comment by Siphonophora** [comment]
@tkohlmeier Could you provide an example?

**Comment by tkohlmeier** [comment]
Of course.

Test class:

    using NUnit.Framework;
    using System;
    using System.Collections.Generic;

    namespace NUnitTestProject1
    {
        public class Tests
        {
            [TestCaseSource(nameof(GetTestCases))]
            public void Test(TestCaseClass testCase)
            {
                Console.WriteLine(testCase);
            }

            public static IEnumerable<TestCaseClass> GetTestCases()
            {
                return new[]
                {
                    new TestCaseClass("a\\b")
                };
            }

            public class TestCaseClass
            {
                public TestCaseClass(string name) => Name = name;

                public string Name { get; set; }

                public override string ToString() => Name;
            }
        }
    }

TestResults.xml:

    <?xml version="1.0" encoding="utf-8"?>
    <test-run id="2" duration="0.002891" testcasecount="1" total="1" passed="1" failed="0" inconclusive="0" skipped="0" result="Passed" start-time="2020-08-31T 18:39:51Z" end-time="2020-08-31T 18:39:52Z">
      <test-suite type="Assembly" name="NUnitTestProject1.dll" fullname="C:\Users\Torben\source\repos\NUnitTestProject1\NUnitTestProject1\bin\Debug\netcoreapp3.1\NUnitTestProject1.dll" total="1" passed="1" failed="0" inconclusive="0" skipped="0" result="Passed" duration="0.002891">
        <test-suite type="TestSuite" name="UnknownNamespace" fullname="UnknownNamespace" total="1" passed="1" failed="0" inconclusive="0" skipped="0" result="Passed" duration="0.002891">
          <test-suite type="TestFixture" name="UnknownType" fullname="UnknownNamespace.UnknownType" total="1" passed="1" failed="0" inconclusive="0" skipped="0" result="Passed" duration="0.002891">
            <test-case name="Test(a\b)" fullname="UnknownNamespace.UnknownType.NUnitTestProject1.Tests.Test(a\b)" methodname="NUnitTestProject1.Tests.Test(a\b)" classname="UnknownType" result="Passed" duration="0.002891" asserts="0">
              <output><![CDATA[a\b

    ]]></output>
            </test-case>
          </test-suite>
        </test-suite>
        <errors />
      </test-suite>
    </test-run>

**Comment by Siphonophora** [comment]
Ok, great. I just gave this a try, which produced the warning below. 

`Unable to parse the test name 'UnitTestProject2.Tests.Test(a\b)' into a type and method. Using Type='UnknownType' and Method='UnitTestProject2.Tests.Test(a\b)'`

I'm not surprised this is an error. When I wrote the name parser, I didn't know of a way to have a `\` that wasn't in quotes. I will try to take a look soon to check if fixing this will break any of the existing tests.
